File: jscomp/test/gpr_2633_test.ml
type foo

external on :
  foo ->
  ([`bar of unit -> unit
   | `foo of string -> unit ] [@bs.string]) ->
   unit = "" [@@bs.send]

let on1 foo event =
  on foo event

(* FIXME *)
let on2 foo h event =
  on foo (h event)

File: jscomp/test/gpr_2633_test.js
'use strict';

var Curry = require("../../lib/js/curry.js");

function on1(foo, $$event) {
  foo.on($$event.NAME, $$event.VAL);
  
}

function on2(foo, h, $$event) {
  foo.on(Curry._1(h, $$event).NAME, Curry._1(h, $$event).VAL);
  
}

exports.on1 = on1;
exports.on2 = on2;
/* No side effect */
